src/highlights/v0.24.md

```diff
@@ -0,0 +1,300 @@
+        width: 240,
+        height: 30,
+    },
+    pixels: Size {
+        width: 0,
+        height: 0,
+    },
+}
+```
+
+With this new API, the goal is to improve image handling in terminal emulators
+by sharing terminal size and layout information,
+enabling precise image placement and resizing within rectangles.
+
+See the pull request for more information:
+[https://github.com/ratatui-org/ratatui/pull/276](https://github.com/ratatui-org/ratatui/pull/276)
+
+## BarChart: Render _smol_ charts 📊
+
+We had a bug where the `BarChart` widget doesn't render labels that are full width.
+Now this is fixed and we are able to render charts smaller than 3 lines!
+
+For example, here is how `BarChart` is rendered and resized from a single line to 4 lines in order:
+
+```
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+
+
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+a b c d e f g h i
+
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+a b c d e f g h i
+      Group
+
+          ▂ ▄ ▆ █
+  ▂ ▄ ▆ 4 5 6 7 8
+a b c d e f g h i
+      Group
+```
+
+If you set `bar_width(2)` then it is rendered as:
+
+```
+          ▂ ▄ ▆ █
+  ▂ ▄ ▆ 4 5 6 7 8
+a b c d e f g h i
+      Group
+```
+
+## Block: custom symbols for borders 🛡️
+
+The `Block` widget has a new method called `border_set` that can be used to specify the symbols that are
+going to be used for the borders.
+
+```rs
+Block::default()
+    .borders(Borders::ALL)
+    .border_set(border::Set {
+        top_left: "1",
+        top_right: "2",
+        bottom_left: "3",
+        bottom_right: "4",
+        vertical_left: "L",
+        vertical_right: "R",
+        horizontal_top: "T",
+        horizontal_bottom: "B",
+    })
+```
+
+When rendered:
+
+```
+1TTTTTTTTTTTTT2
+L             R
+3BBBBBBBBBBBBB4
+```
+
+There are also 2 new border types added (`QuadrantInside`, `QuadrantOutside`).
+
+See the available border types at <https://docs.rs/ratatui/latest/ratatui/widgets/block/enum.BorderType.html>
+
+Also, there are breaking changes to note here:
+
+-   `BorderType::to_line_set` is renamed to `to_border_set`
+-   `BorderType::line_symbols` is renamed to `border_symbols`
+
+## Canvas: half block marker 🖼️
+
+A new marker named `HalfBlock` is added to `Canvas` widget along with the associated
+`HalfBlockGrid`.
+
+The idea is to use half blocks to draw a grid
+of "pixels" on the screen. Because we can set two colors per cell, and
+because terminal cells are about twice as tall as they are wide, we can
+draw a grid of half blocks that looks like a grid of square pixels.
+
+```rs
+Canvas::default()
+    .marker(Marker::HalfBlock)
+    .x_bounds([0.0, 10.0])
+    .y_bounds([0.0, 10.0])
+    .paint(|context| {
+        context.draw(&Rectangle {
+            x: 0.0,
+            y: 0.0,
+            width: 10.0,
+            height: 10.0,
+            color: Color::Red,
+        });
+    });
+```
+
+Rendered as:
+
+```
+█▀▀▀▀▀▀▀▀█
+█        █
+█        █
+█        █
+█        █
+█        █
+█        █
+█        █
+█        █
+█▄▄▄▄▄▄▄▄█
+```
+
+## Frame: no more generics 🚫
```

Comment:
Since every user that tries to upgrade is going to be affected by this, it might be good to add `BREAKING CHANGE` to title and/or move this section to the top of the file.

Reply:
moved it after website section in 4aaf6f5671ddd5930500e818c4a4e7e051c430c0

should be visible enough 🤔